src/main/java/com/github/wovnio/wovnjava/Headers.java

```diff
@@ -111,29 +111,8 @@
         }
         this.url += this.removeLang(this.query, this.langCode());
         this.url = this.url.length() == 0 ? "/" : this.url;
-        if (this.settings.query.size() > 0) {
-            ArrayList<String> queryVals = new ArrayList<String>();
-            for (String q : this.settings.query) {
-                Pattern p = Pattern.compile("(^|&)(" + q + "[^&]+)(&|$)");
-                Matcher m = p.matcher(this.query);
-                if (m.find() && m.group(2) != null && m.group(2).length() > 0) {
-                    queryVals.add(m.group(2));
-                }
-            }
-            if (queryVals.isEmpty()) {
-                // ignore all query parameters.
-                this.query = "";
-            } else {
-                this.query = "?";
-                Collections.sort(queryVals);
-                for (String q : queryVals) {
-                    this.query += q + "&";
-                }
-                // remove last ampersand.
-                this.query = Pattern.compile("&$").matcher(this.query).replaceFirst("");
-            }
-        } else {
-            this.query = "";
+        if (!this.query.isEmpty() && !this.query.startsWith("?")) {
+            this.query = "?" + this.query;
```

Comment:
Note that this modification is required to match the expected format that resulted from the old code. It is not elegant, but `this.query` should be considered a legacy part of `Headers.java`, and will be phased out soon.